proof/refine/Invariants_H.thy

```diff
@@ -2695,6 +2695,14 @@ lemma obj_at'_ko_at'_prop:
   "ko_at' ko t s \<Longrightarrow> obj_at' P t s = P ko"
   by (drule obj_at_ko_at', clarsimp simp: obj_at'_def)
 
+lemma ko_wp_at'_not_o:
+  "ko_wp_at' (\<lambda>a. \<not> P a) t s = ko_wp_at' (Not o P) t s"
```

Comment:
Not a fan of these: \<circ> is newer than `o`, and also it's fun_comp or comp, the suffix "not_o" is not guessable, and on top of that this is an equality lemma. If you look at some of the lemmas above, it doesn't blend in at all.

Reply:
I'd call this one `ko_wp_at'_not_comp_fold`

Reply:
These were based on abstract versions for `obj_at`, which is I think where the name and use of `o` instead of `comp` and `<circ>` came from. I'll update them.

With regards to the location, I originally had them where I needed them and wanted to move them somewhere up earlier. This was the best place that I could quickly find in Invariants_H but I'm open to suggestions.